pkg/R/RDD.R

```diff
@@ -1382,26 +1382,18 @@ setMethod("groupByKey",
             groupVals <- function(part) {
               vals <- new.env()
               keys <- new.env()
+              pred <- function(item) exists(item$hash, keys)
               # Each item in the partition is list of (K, V)
               lapply(part,
                      function(item) {
-                       hashVal <- as.character(hashCode(item[[1]]))
-                       if (exists(hashVal, vals)) {
-                         acc <- vals[[hashVal]]
-                         acc[[length(acc) + 1]] <- item[[2]]
-                         vals[[hashVal]] <- acc
-                       } else {
-                         vals[[hashVal]] <- list(item[[2]])
-                         keys[[hashVal]] <- item[[1]]
-                       }
+                       item$hash <- as.character(hashCode(item[[1]]))
+                       updateOrCreatePair(item, keys, vals, pred,
+                                          function(vs, v) c(vs, list(v)),
```

Comment:
Can we emulate the old way by using [`addItemToAccumulator `](https://github.com/amplab-extras/SparkR-pkg/blob/25639cf25182a2f93e0ff7c3a76ca2844da0d29b/pkg/R/utils.R#L181)? Note that there's also `initAccumulator` which might be useful.